docs/api.md

```diff
@@ -0,0 +1,371 @@
+- [Cloud Hypervisor API](#cloud-hypervisor-api)
+  * [External API](#external-api)
+    + [REST API](#rest-api)
+      - [Location and availability](#location-and-availability)
+      - [Endpoints](#endpoints)
+		* [Virtual Machine Manager (VMM) Actions](#virtual-machine-manager--vmm--actions)
+		* [Virtual Machine (VM) Actions](#virtual-machine--vm--actions)
+      - [REST API Examples](#rest-api-examples)
+        * [Create a Virtual Machine](#create-a-virtual-machine)
+	    * [Boot a Virtual Machine](#boot-a-virtual-machine)
+        * [Dump a Virtual Machine Information](#dump-a-virtual-machine-information)
+        * [Reboot a Virtual Machine](#reboot-a-virtual-machine)
+        * [Shut a Virtual Machine Down](#shut-a-virtual-machine-down)
+    + [Command Line Interface](#command-line-interface)
+    + [REST API and CLI Architecural Relationship](#rest-api-and-cli-architecural-relationship)
```

Comment:
Typo:
~~`[REST API and CLI Architecural Relationship](#rest-api-and-cli-architecural-relationship)`~~
`[REST API and CLI Architectural Relationship](#rest-api-and-cli-architectural-relationship)`

Reply:
:-/ Thanks for the thorough review @likebreath 
It should be clean now...